image_transport/include/image_transport/publisher_plugin.hpp

```diff
@@ -135,10 +160,11 @@ class PublisherPlugin
    * \brief Advertise a topic. Must be implemented by the subclass.
    */
   virtual void advertiseImpl(
-    rclcpp::Node * node,
     const std::string & base_topic,
     rmw_qos_profile_t custom_qos,
     rclcpp::PublisherOptions options) = 0;
+
+  std::shared_ptr<NodeType> node_;
```

Comment:
by holding a reference to the Node here, are we going to end up with some sort of hanging reference where the node never goes out of scope? (I don't recall the specifics, but this was a big problem for Nav2 and they eventually switched to storing WeakPtrs inside the various plugins so that the SharedPtr would eventually go out of scope as expected)

Reply:
I'll change them into weak pointers, does that sound good to you?

Reply:
Fixed as these are not needed anymore.